### PR TITLE
Added token replacement to removal

### PR DESCRIPTION
**Pull request by MrMCake**
# Change

- Added token replacement to removal

Pipeline reference
[AnalysisServices: Servers](https://github.com/Azure/ResourceModules/actions/workflows/ms.analysisservices.servers.yml)
[Compute: VirtualMachines](https://github.com/Azure/ResourceModules/actions/workflows/ms.compute.virtualmachines.yml)

## Type of Change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update (Wiki)

### CODE REVIEW

File: utilities/pipelines/resourceRemoval/helper/Get-DependencyResourceNameList.ps1 [hunk 1]
```diff
@@ -19,15 +19,30 @@ function Get-DependencyResourceNameList {
...
     $parameterFolders = Get-ChildItem -Path $dependencyParameterPath -Recurse -Filter 'parameters' -Directory
     $parameterFilePaths = [System.Collections.ArrayList]@()
     foreach ($parameterFolderPath in $parameterFolders.FullName) {
         $parameterFilePaths += Get-ChildItem -Path $parameterFolderPath -Recurse -Filter '*.json'
     }
 
+    $Settings = Get-Content -Path (Join-Path $repoRootPath 'settings.json') | ConvertFrom-Json
+    foreach ($parameterFilePath in $parameterFilePaths) {
+        $ConvertTokensInputs = @{
+            ParameterFilePath              = $parameterFilePath
```
Comment (eriqua) [inline utilities/pipelines/resourceRemoval/helper/Get-DependencyResourceNameList.ps1:38]: Which parameter file would this be? Will all dependency parameter files be converted?

File: utilities/pipelines/resourceRemoval/helper/Get-DependencyResourceNameList.ps1 [hunk 2]
```diff
@@ -19,15 +19,30 @@ function Get-DependencyResourceNameList {
...
+    $repoRootPath = (Get-Item $PSScriptRoot).Parent.Parent.Parent.Parent.FullName
+    . (Join-Path $repoRootPath 'utilities' 'pipelines' 'tokensReplacement' 'Convert-TokensInParameterFile.ps1')
+
     $parameterFolders = Get-ChildItem -Path $dependencyParameterPath -Recurse -Filter 'parameters' -Directory
     $parameterFilePaths = [System.Collections.ArrayList]@()
     foreach ($parameterFolderPath in $parameterFolders.FullName) {
         $parameterFilePaths += Get-ChildItem -Path $parameterFolderPath -Recurse -Filter '*.json'
     }
 
+    $Settings = Get-Content -Path (Join-Path $repoRootPath 'settings.json') | ConvertFrom-Json
```
Comment (eriqua) [inline utilities/pipelines/resourceRemoval/helper/Get-DependencyResourceNameList.ps1:35]:
```suggestion
    # Replace tokens in parameter file
    $Settings = Get-Content -Path (Join-Path $repoRootPath 'settings.json') | ConvertFrom-Json
```

File: utilities/pipelines/resourceRemoval/helper/Get-DependencyResourceNameList.ps1 (hunk 1 above)
Comment (MrMCake) [reply utilities/pipelines/resourceRemoval/helper/Get-DependencyResourceNameList.ps1:38]: Yes, all

**Review by eriqua — APPROVED** [review]
🆗

**Pull request merged**